theforeman.org/scripts/debian/stage_packages.sh

```diff
@@ -7,19 +7,46 @@ set -xe
 # * the web node must have the freight class applied
 
 # Find the deps in the build-dir from the previous step
-DEB_PATH=./debian/${os}/build-${project}
+for possible_path in ./debian/${os}/build-${project} ./plugins/build-${project} ./dependencies/${os}/build-${project}; do
+  if [[ -d "${possible_path}" ]]; then
+    DEB_PATH="${possible_path}"
+  fi
+done
 
 # Upload all builds to stagingdeb for testing
-echo "scratch build: uploading to stagingdeb/${os}/${repoowner}-${version}"
+HOSTS="web01.osuosl.theforeman.org"
 export RSYNC_RSH="ssh -i /home/jenkins/workspace/staging_key/rsync_freightstage_key"
 USER=freightstage
-HOSTS="web01.osuosl"
-COMPONENT=${repoowner}-${version}
+SUITE=${os}
+
+if [[ "$DEB_PATH" == *"plugins"* ]];then
+  SUITE="plugins"
+
+  # Plugins aren't staged.
```

Comment:
@evgeni Why are plugins not staged?

Reply:
For the same reason they aren't in RPM land? We don't want one plugin to hold all the plugins. (Plus we don't have gating tests for most plugins anyways)

Reply:
Technically, they are staged in RPM land, as we mash the repository into a staging repository on Koji, then we perform repoclosure and then we promote the repository to production location.

Reply:
As we don't perform any repoclosure on Debian… ;)

Reply:
I get that part, I was more wanting to understand if it was a technical reason or a process reason. Different processes make automating, and understanding harder. What would be the technical downside of staging them first? Time and space?

Reply:
Space is probably negligible. Time… Well, it takes 10-ish minutes to publish Debian, so we'd add that time somewhere (unless we add the unstaging to the normal Foreman pipe).
So, what would unstage them? A normal Foreman pipeline? A dedicated pipeline that does effectively nothing?